doc/faq.mld

```diff
@@ -0,0 +1,103 @@
+But it is better not to set individual options to override what the default profile is doing.
+
+To quote (and sed) {{:https://prettier.io/docs/en/option-philosophy.html}prettier's page on option philosophy}:
+
+{%html:
+<q>
+OCamlFormat has a few options because of history. <i>But we don’t want more of them.</i>
+
+By far the biggest reason for adopting OCamlFormat is to stop all the on-going debates over styles.
+
+The more options OCamlFormat has, the further from the above goal it gets. The debates over styles just turn into debates over which OCamlFormat options to use.
+</q>
+%}
+
+{2 Can OCamlFormat handle Reason files?}
+
+OCamlFormat is influenced by and follows the same basic design as [refmt] for {{:https://github.com/facebook/reason}Reason}, but outputs OCaml instead of Reason.
+
+This tool is not able to deal directly with Reason code ([*.re]/[*.rei] files),
+but it is possible to first convert these files to OCaml syntax using [refmt -p
+ml] and then running [ocamlformat] on this output.
+
+{2 How to ignore directories?}
+
+It is possible to disable OCamlFormat for the files of a directory by having an [.ocamlformat] file containing disable in this directory, or listing the files to ignore in an [.ocamlformat-ignore] file.
+
+For now it is not possible to recursively ignore all subdirectories and files from a parent directory, you need to list all the descendants of the directory to ignore them, e.g.:
+{[
+dir/*
+dir/*/*
+dir/*/*/*
+]}
+
+{2 How to locally disable OCamlFormat?}
+
+To disable the formatting of a specific toplevel item you must attach an [[@@ocamlformat "option=VAL"]] attribute to this item in the processed file, such as:
+
+{[
+let do_not_touch
+    (x : t)
+      (y : t)
+        (z : t) = [
+  x; y; z
+] [@@ocamlformat "disable"]
+]}
+
+To disable the formatting of a specific expression you must attach an [[@ocamlformat "option=VAL"]] attribute to this expression in the processed file, such as:
+
+{[
+let do_not_touch (x : t) (y : t) (z : t) = [
+  x; y; z
+] [@ocamlformat "disable"]
+]}
+
+To disable a whole file, the preferred way is to add the name of the file to a local [.ocamlformat-ignore] file. An [.ocamlformat-ignore] file specifies files that OCamlFormat should ignore. Each line in an [.ocamlformat-ignore] file specifies a filename relative to the directory containing the [.ocamlformat-ignore] file.
+Shell-style regular expressions are supported. Lines starting with [#] are ignored and can be used as comments.
+
+{2 Why is there an inconsistent spacing on the | character in match blocks?}
+
+The change in the indentation of [| ] vs [|] would indicate that there is a short body of a nested or-pattern that could be easily missed if formatted as: [| ]. This can be disabled using [indicate-nested-or-patterns = false] if you are happy with the risk. Alternatively, [break-cases = nested] can be used to break after the [->] to avoid the risk at the cost of significantly less compact code.
```

Comment:
```suggestion
The change in the indentation of [| ] vs [|] would indicate that there is a short body of a nested or-pattern that could be easily missed if formatted as: [| ]. This can be disabled using [indicate-nested-or-patterns = unsafe-no] if you are happy with the risk. Alternatively, [break-cases = nested] can be used to break after the [->] to avoid the risk at the cost of significantly less compact code.
```

The default is already `unsafe-no`, so I think it's reasonable to not say it's a risk when we already decided it is fine.